pkg/reconciler/app/reconciler.go

```diff
@@ -247,3 +250,59 @@ func (r *Reconciler) updateStatus(desired *v1alpha1.App) (*v1alpha1.App, error)
 
 	return r.KfClientSet.KfV1alpha1().Apps(existing.GetNamespace()).UpdateStatus(existing)
 }
+
+// gcRevisions is necessary because Knative won't scale down revisions
+// that have a `minScale` greater than 0.
+// TODO: Reevaluate once https://github.com/knative/serving/issues/4183 is
+// resolved.
+func (r *Reconciler) gcRevisions(ctx context.Context, app *v1alpha1.App) error {
+	r.Logger.Infof("Checking for revisions that need to adjust %s...", autoscaling.MinScaleAnnotationKey)
+	defer r.Logger.Infof("Done checking for revisions that need to adjust %s.", autoscaling.MinScaleAnnotationKey)
+
+	if app.Status.LatestCreatedRevisionName != app.Status.LatestReadyRevisionName {
+		r.Logger.Infof("Not willing to GC revisions while the latest is not ready...")
+		return nil
+	}
+
+	selector := labels.Set{"serving.knative.dev/configuration": app.Name}.AsSelector()
+	revs, err := r.knativeRevisionLister.Revisions(app.Namespace).List(selector)
+	if err != nil {
+		return err
+	}
+
+	if len(revs) == 0 {
+		return nil
+	}
+
+	revisionClient := r.ServingClientSet.ServingV1alpha1().Revisions(app.Namespace)
+
+	parseGeneration := func(rev *serving.Revision) int64 {
+		v, ok := rev.Labels["serving.knative.dev/configurationGeneration"]
+		if !ok {
+			r.Logger.Warnf("Revision did not contain ConfigurationGeneration")
+			return -1
+		}
+
+		x, err := strconv.ParseInt(v, 10, 64)
+		if err != nil {
+			r.Logger.Warnf("Revision had an invalid ConfigurationGeneration: %s", err)
+			return -1
+		}
+		return x
+	}
+
+	// descending
+	sort.Slice(revs, func(i int, j int) bool {
+		return parseGeneration(revs[j]) < parseGeneration(revs[i])
+	})
+
+	// delete everything after the latest generation
+	for _, rev := range revs[1:] {
+		r.Logger.Infof("GC'ing Revision %s...", rev.Name)
```

Comment:
Consider making this the only info log message and the rest of them debug messages. In a large cluster these messages might become spammy. 

Also can you replace GC'ing with Garbage collecting? I think GC made sense in the context of the other messages but not so much by itself.

Reply:
Done